### PR TITLE
Add core tab support for checksums

### PR DESCRIPTION
**Pull request by NikhilBartwal**
PR in reference to Issue #2420
`core/download/checksums.py' has been fixed to now use `tabs` for storage and retrieval of checksums

### CODE REVIEW

**Review by Conchylicultor — CHANGES_REQUESTED** [review]
Thank you, but this will fail for legacy checksums files.
Could you make sure this is backward compatible with old checksum files which still uses space ?

**Comment by NikhilBartwal** [comment]
@Conchylicultor Sorry for that, that part had been left out by mistake while creating different branches :(

**Review by Conchylicultor — APPROVED** [review]
Thank you!

File: tensorflow_datasets/core/download/checksums.py [hunk 1]
```diff
@@ -151,16 +151,14 @@ def _parse_url_infos(checksums_file: Iterable[str]) -> Dict[str, UrlInfo]:
...
-    url, size, checksum = line.rsplit(' ', 2)
+    try:
+      url, size, checksum = line.rsplit('\t', 2)
+    except ValueError:
+      url, size, checksum = line.rsplit(' ', 2)
     url_infos[url] = UrlInfo(size=int(size), checksum=checksum)
   return url_infos
 
 
-def url_infos_from_path(checksums_path: str) -> Dict[str, UrlInfo]:
```
Comment (Conchylicultor) [inline tensorflow_datasets/core/download/checksums.py:159]: Note that this function is called in other part of the code so this is breaking TFDS. I've removed this changed internally.

Comment (NikhilBartwal) [reply tensorflow_datasets/core/download/checksums.py:159]: @Conchylicultor Do i need to change the PR or is it alright?

Comment (Conchylicultor) [reply tensorflow_datasets/core/download/checksums.py:159]: No need to update this PR

**Pull request merged**